File: 백준/Silver/15810. 풍선 공장/풍선 공장.cc
#include <iostream>
#include <vector>
using namespace std;
int main(void) {
    int n, m;
    long long int start, end, mid, s, r = 0, maxn = 0;
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cin >> n >> m;
    vector<int> arr(n);
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
        if (maxn < arr[i])
            maxn = arr[i];
    }
    start = 0, end = maxn * m;
    while (start <= end) {
        mid = (start + end) / 2;
        s = 0;
        for (int i = 0; i < n; i++)
            s += mid / arr[i];
        if (s < m)
            start = mid + 1;
        else {
            end = mid - 1;
            r = mid;
        }
    }
    cout << r;
    return 0;
}

File: 백준/Silver/15810. 풍선 공장/풍선 공장.py
import sys
input=lambda:sys.stdin.readline().rstrip()
mii=lambda:map(int,input().split())
n,m=mii()
arr=[*mii()]
start,end=1,max(arr)*m
while start<=end:
    mid=(start+end)//2
    s=0
    for i in arr:
        s+=mid//i
    if s<m:
        start=mid+1
    else:
        end=mid-1
        result=mid
print(result)